File: BHAVRP_Java/BHAVRPv2.0/BHAVRP2.0/src/cujae/inf/ic/om/problem/solution/Cluster.java
package cujae.inf.ic.om.problem.solution;

import java.util.ArrayList;

public class Cluster {

	private int idCluster;
	private double requestCluster;
	private ArrayList<Integer> itemsOfCluster;
	
	public Cluster() {
		super();
		idCluster = -1;
		this.requestCluster = 0.0;
		this.itemsOfCluster = new ArrayList<Integer>();	
	}

	public Cluster(int idCluster, double requestCluster,
			ArrayList<Integer> itemsOfCluster) {
		super();
		this.idCluster = idCluster;
		this.requestCluster = requestCluster;
		this.itemsOfCluster = itemsOfCluster;
	}

	public int getIDCluster() {
		return idCluster;
	}

	public void setIDCluster(int idCluster) {
		this.idCluster = idCluster;
	}

	public ArrayList<Integer> getItemsOfCluster() {
		return itemsOfCluster;
	}

	public void setItemsOfCluster(ArrayList<Integer> itemsOfCluster) {
		this.itemsOfCluster = itemsOfCluster;
	}

	public double getRequestCluster() {
		return requestCluster;
	}

	public void setRequestCluster(double requestCluster) {
		this.requestCluster = requestCluster;
	}

	 public void cleanCluster() {
		 requestCluster = 0.0;	
		 itemsOfCluster.clear();	 
    }
}

File: BHAVRP_Python/cujae/inf/citi/om/problem/output/cluster.py
from typing import List
from dataclasses import dataclass, field

@dataclass
class Cluster:
    id_cluster: int = -1
    request_cluster: float = 0.0
    items_of_cluster: List[int] = field(default_factory=list)
    
    def get_id_cluster(self) -> int:
        return self.id_cluster
    
    def set_id_cluster(self, value: int):
        self.id_cluster = value
        
    def get_request_cluster(self) -> float:
        return self.request_cluster
    
    def set_request_cluster(self, value: float):
        self.request_cluster = value
        
    def get_items_of_cluster(self) -> List[int]:
        return self.items_of_cluster

    def set_items_of_cluster(self, value: List[int]):
        self.items_of_cluster = value
            
    # Limpia el cluster reiniciando la solicitud y vaciando la lista de elementos.       
    def clean_cluster(self):
        self.request_cluster = 0.0
        self.items_of_cluster.clear()